friends/src/Form.js

```diff
@@ -0,0 +1,160 @@
+import React, { Fragment, Component } from 'react';
+import axios from 'axios';
+import { Route } from 'react-router-dom';
+import styled from 'styled-components';
+
+const FormDiv = styled.form`
+    background-color: #f3eff6
+    margin-right: 5rem;
+    border: 1px solid gray;
+    border-radius: 4px;
+    margin-top: 5rem;
+    width: 300px;
+    height: 375px;
+    display: flex;
+    flex-direction: column;
+    padding: 2rem 3.5rem;
+    justify-content: space-evenly;
+`
+
+const H2 = styled.h2`
+    font-size: 2rem;
+    margin: 3rem 0;
+`
+
+const Input = styled.input`
+    padding: .75rem 1.5rem;
+    font-size: 1.5rem;
+    border-radius: 4px;
+    border: 1px solid lightgray;
+    &:focus {
+        outline: none;
+    }
+`
+
+const Btn = styled.button`
+    background-color: #957aae;
+    border: none;
+    font-size: 1.4rem;
+    color: #fff;
+    padding: 1rem;
+    margin-bottom: 5rem;
+    border-radius: 4px;
+    &:focus {
+        outline: none;
+    }
+    &:hover {
+        cursor: pointer;
+        font-weight: bold;
+    }
+`
+
+export default function Form(props) {
+    return (
+        <Fragment>
+            <Route exact path="/" render={routeProps => <FormAdd {...routeProps} updateFriends={props.updateFriends} />} />
+            <Route path="/update/:id" render={routeProps => <FormUpdate {...routeProps} updateFriends={props.updateFriends} />} />
+        </Fragment>
+    );
+}
+
+
+class FormUpdate extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            nameInput: null,
+            ageInput: null,
+            emailInput: null,
+        }
+    }
+
+    changeHandler = (e) => {
+        this.setState({
+            ...this.state,
+            [e.target.name]: e.target.value
+        })
+    }
+
+    submitHandler = (e) => {
+        e.preventDefault();
+            axios
+                .put(`http://localhost:5000/friends/${this.props.match.params.id}`, {
+                    name: `${this.state.nameInput}`,
+                    age: this.state.ageInput,
+                    email: this.state.emailInput
+                })
+                .then(this.setState({
+                    nameInput: '',
+                    ageInput: '',
+                    emailInput: '',
+                }))
+                .then(this.props.updateFriends)
+                .then(this.props.history.push('/'))
```

Comment:
This isn't a function; I'm not sure if the behavior changes very much in this case, but this is a case where you might prefer adding in an extra step since it's not a function but an invocation:
```javascript
.then(() => this.props.history.push('/'))
```